src/frontend/components/UI/AuthModal/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef } from 'react'
+import styles from './index.module.scss'
+import { ModalAnimation } from '@hyperplay/ui'
+import { WebviewTag } from 'electron'
+import { observer } from 'mobx-react-lite'
+import authModalState from '../../../state/authModalState'
+import { ethers } from 'ethers'
+import extensionStore from '../../../store/ExtensionStore'
+import onboardingStore from '../../../store/OnboardingStore'
+
+// TODO: replace this with dev portal prod URL when it's ready
+const url =
+  'https://hyperplay-dev-git-feature-unified-auth-ui-valist.vercel.app/signin'
+
+const METAMASK_ALREADY_PROVIDED_ERROR_CODE = -32002
+
+const isTooManyRequestsError = (error: string) => {
+  return error.includes(METAMASK_ALREADY_PROVIDED_ERROR_CODE.toString())
+}
+
+const AuthModal = () => {
```

Comment:
thoughts about moving some of this component state into the `authModalState` mobx class? for achievements, we may want to observe an `isHyperPlayAccountSignedIn` from `authModalState` to update ui elements. Or a variable indicated which stores are connected to the account